core/server/worker/src/main/java/alluxio/worker/dora/RocksDBDoraMetaStore.java

```diff
@@ -0,0 +1,187 @@
+            .setCreateMissingColumnFamilies(true)
+            .setCreateIfMissing(true)
+            .setMaxOpenFiles(-1);
+    columns.add(new ColumnFamilyDescriptor(DORA_META_FILE_STATUS_COLUMN.getBytes(),
+            new ColumnFamilyOptions()
+                    .setMemTableConfig(new HashLinkedListMemTableConfig())
+                    .setCompressionType(CompressionType.NO_COMPRESSION)));
+    mToClose.addAll(columns.stream().map(
+            ColumnFamilyDescriptor::getOptions).collect(Collectors.toList()));
+
+    mRocksStore = new RocksStore(DORA_META_STORE_NAME, dbPath, backupPath, opts, columns,
+            Arrays.asList(mFileStatusColumn), false);
+  }
+
+  /**
+   * Queries the metadata for a file.
+   *
+   * @param path the full path of this file
+   * @return the metadata if found, Optional.empty if not found
+   */
+  @Override
+  public Optional<DoraMeta.FileStatus> getDoraMeta(String path) {
+    byte[] status;
+    try {
+      status = db().get(mFileStatusColumn.get(), path.getBytes());
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+    if (status == null) {
+      return Optional.empty();
+    }
+    try {
+      return Optional.of(DoraMeta.FileStatus.parseFrom(status));
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Stores the metadata identified by URI of the file into this RocksDB.
+   *
+   * @param path the full path of this file
+   * @param meta the block metadata
+   */
+  @Override
+  public void putDoraMeta(String path, DoraMeta.FileStatus meta) {
+    try {
+      db().put(mFileStatusColumn.get(), mWriteOption, path.getBytes(),
+              meta.toByteString().toByteArray());
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Removes the metadata from the RocksDB. It is not an error if the record specified by the key
+   * is not found.
+   *
+   * @param path the full path of the file whose metadata to be removed
+   */
+  @Override
+  public void removeDoraMeta(String path) {
+    try {
+      db().delete(mFileStatusColumn.get(), mWriteOption, path.getBytes());
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Clears all the metadata in this RocksDB.
+   */
+  @Override
+  public void clear() {
+  }
+
+  /**
+   * Closes the RocksDb and all resources.
+   */
+  @Override
+  public void close() {
+    LOG.info("Closing " + DORA_META_DB_NAME + " and recycling all RocksDB JNI objects");
+    // Close the elements in the reverse order they were added
+    Collections.reverse(mToClose);
+    mToClose.forEach(RocksObject::close);
+    mRocksStore.close();
+    LOG.info(DORA_META_DB_NAME + " closed");
+  }
+
+  /**
+   * Queries the estimated number of the records in this RocksDB.
+   * Please note, this is not an accurate number.
+   *
+   * @return the estimated number of records
+   */
+  @Override
+  public long size() {
```

Comment:
Optional<Long> ?

Reply:
Do you mean the result is Optional?
Or this interface is Optional?

The result is a long integer, not Optional.

Reply:
Yeah, becuase size 0 can refer to it's nothing in metastore or it's failed to get the size, so Optional<Long> of size() gives distinguish between those two cases and is the convention we want to follow in the future

Reply:
Fixed.